ripc-protocol-tcp/src/main/java/io/ripc/protocol/tcp/TcpInterceptor.java

```diff
@@ -0,0 +1,7 @@
+package io.ripc.protocol.tcp;
+
+public interface TcpInterceptor<I, O, II, OO> {
+
+    TcpHandler<II, OO> intercept(TcpHandler<I, O> handler);
```

Comment:
I was envisioning interceptors being analogous to Netty's pipeline feature. Pipelines can be dynamically altered and new ones are created when new connections are made. How would we use this intercept mechanism to wrap each new connection dynamically (especially those that come in as a result of an automatic reconnect after a client connection has died)?